packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -2756,6 +2756,23 @@ class ScaffoldState extends State<Scaffold> with TickerProviderStateMixin, Resto
             _dismissedBottomSheets.remove(bottomSheet);
           });
         }
+        if (_floatingActionButtonVisibilityValue != 1.0 || _showBodyScrim) {
+          FlutterError.reportError(FlutterErrorDetails(
+            exception: StateError(
+              'A DraggableScrollableSheet was used as a BottomSheet and '
```

Comment:
Is DraggableScrollableSheet the only widget that can make this error possible? What is the root cause? I wonder if folks might get this error message without using that widget and be very confused.

Reply:
Well, since this is causing other tests to fail, probably not haha.

I'll try to look at this some more and figure out what's going on.

Reply:
Ok - I had a bug in my check, if there's no FAB then the value is 0.

However, there's another problem: the `showBodyScrim` method is public on the state, so in theory anyone could call it from anywhere and do what they want.

I don't think this patch really works. Going to close it while I think about this some more.